handlers/postitem/postItem.go

```diff
@@ -0,0 +1,70 @@
+package postitem
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"regexp"
+	"strings"
+
+	"github.com/aws/aws-lambda-go/events"
+	"github.com/mount-joy/thelist-lambda/data"
+	"github.com/mount-joy/thelist-lambda/db"
+	"github.com/mount-joy/thelist-lambda/handlers/iface"
+)
+
+type postItem struct {
+	db db.DB
+}
+
+// New returns an instance of postItem satisfying the RouteHandler interface
+func New() iface.RouteHandler {
+	return &postItem{
+		db: db.DynamoDB(),
+	}
+}
+
+// Match returns true if this RouteHandler should handle this request
+func (g *postItem) Match(request events.APIGatewayV2HTTPRequest) bool {
+	var re = regexp.MustCompile(`^/lists/([\w-]+)/items/?$`)
+	return request.RequestContext.HTTP.Method == "POST" && re.MatchString(request.RequestContext.HTTP.Path)
+}
+
+// Handle handles this request and returns the response and status code
+func (g *postItem) Handle(request events.APIGatewayV2HTTPRequest) (interface{}, int) {
+	listID, err := getListID(request.RequestContext.HTTP.Path)
+	if err != nil {
+		log.Printf("Error: %s", err.Error())
```

Comment:
just fyi is you used `%v` you could just use `err`
I haven't tested but what you have might actually be faster

Reply:
I don't really like the logging in general as it doesn't have line numbers, we should fix it at some point

Reply:
Yeah we can definitely do better. Can do all of it in a single PR